### PR TITLE
[SYCL] empty event constructor, more explicit host event handling

### PR DESCRIPTION
**Pull request by cperkinsintel**
making host events more explicit. simplifies the logic, makes checking for host (or not) simpler, and reduces potential contention.  This is a follow up to https://github.com/intel/llvm/pull/6296

Signed-off-by: Chris Perkins <chris.perkins@intel.com>

### CODE REVIEW

File: sycl/source/detail/event_impl.cpp [hunk 1]
```diff
@@ -126,6 +128,10 @@ void event_impl::setContextImpl(const ContextImplPtr &Context) {
   MIsContextInitialized = true;
 }
 
+event_impl::event_impl(bool)
+    : MIsInitialized(false), MHostEvent(false), MIsFlushed(true),
+      MState(HES_Complete) {}
```
Comment (steffenlarsen) [inline sycl/source/detail/event_impl.cpp:133]: Could this be combined with the `event_impl(HostEventState State = HES_Complete)` constructor into something like `event_impl(HostEventState State = HES_Complete, bool IsHostEvent = true)`?

Comment (cperkinsintel) [reply sycl/source/detail/event_impl.cpp:133]: No. We need a new constructor for event_impl that'll create non-host events that the argumentless event() constructor can call.   Meanwhile, we leave in place the (mostly host) temporary event_impl that the scheduler constructs.

Comment (steffenlarsen) [reply sycl/source/detail/event_impl.cpp]: This was discussed offline and an alternative was found.

**Comment by pvchupin** [comment]
@cperkinsintel, please address post-commit issue on Windows: https://github.com/intel/llvm/runs/7206588798?check_suite_focus=true

**Pull request merged**